packages/components/src/components/multi-select/multi-select-option/multi-select-option.tsx

```diff
@@ -0,0 +1,72 @@
+import { Component, Element, h, Host, type JSX, Prop } from '@stencil/core';
+import { AllowedTypes, attachComponentCss, getPrefixedTagNames, validateProps } from '../../../utils';
+import { MultiSelectOptionInternalHTMLProps } from './multi-select-option-utils';
+import type { PropTypes } from '../../../types';
+import { getComponentCss } from './multi-select-option-styles';
+import { getOptionAriaAttributes } from '../../../utils/a11y/select/select-aria';
+
+const propTypes: PropTypes<typeof MultiSelectOption> = {
+  value: AllowedTypes.string,
+  selected: AllowedTypes.boolean,
+  disabled: AllowedTypes.boolean,
+};
+
+@Component({
+  tag: 'p-multi-select-option',
+  shadow: true,
+})
+export class MultiSelectOption {
+  @Element() public host!: HTMLElement & MultiSelectOptionInternalHTMLProps;
+
+  /** The option value. */
+  @Prop() public value: string;
+
+  /** The option selected state. */
+  @Prop({ mutable: true }) public selected?: boolean = false; // eslint-disable-line @typescript-eslint/no-inferrable-types
+
+  /** Disables the option. */
+  @Prop() public disabled?: boolean = false; // eslint-disable-line @typescript-eslint/no-inferrable-types
+
+  /** Hides options which are not matching the searchString **/
+  private hidden = false;
+
+  public componentDidUpdate(): void {
+    this.host.dispatchEvent(
+      new CustomEvent('internalOptionUpdate', {
+        bubbles: true,
```

Comment:
I think this might bubble up until document unless there is a `stopPropagation()` call somewhere?